internal/network/ntp_utils.go

```diff
@@ -0,0 +1,146 @@
+package network
+
+import (
+	"bytes"
+	"context"
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+	"text/template"
+
+	"github.com/go-openapi/swag"
+	"github.com/openshift/assisted-service/internal/common"
+	"github.com/openshift/assisted-service/models"
+	"github.com/openshift/assisted-service/restapi"
+	operations "github.com/openshift/assisted-service/restapi/operations/manifests"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+	"github.com/thoas/go-funk"
+)
+
+//go:generate mockgen -source=ntp_utils.go -package=network -destination=mock_ntp_utils.go
+
+type NtpUtilsAPI interface {
+	AddChronyManifest(ctx context.Context, log logrus.FieldLogger, c *common.Cluster) error
+}
+
+type NtpUtils struct {
+	manifestsApi restapi.ManifestsAPI
+}
+
+func NewNtpUtils(manifestsApi restapi.ManifestsAPI) *NtpUtils {
+	return &NtpUtils{
+		manifestsApi: manifestsApi,
+	}
+}
+
+const defaultChronyConf = `
+driftfile /var/lib/chrony/drift
+makestep 1.0 3
+rtcsync
+logdir /var/log/chrony`
+
+const machineConfigManifest = `
+apiVersion: machineconfiguration.openshift.io/v1
+kind: MachineConfig
+metadata:
+  labels:
+    machineconfiguration.openshift.io/role: master
+  name: masters-chrony-configuration
+spec:
+  config:
+    ignition:
+      config: {}
+      security:
+        tls: {}
+      timeouts: {}
+      version: 2.2.0
+    networkd: {}
+    passwd: {}
+    storage:
+      files:
+      - contents:
+          source: data:text/plain;charset=utf-8;base64,{{.CHRONY_CONTENT}}
+          verification: {}
+        filesystem: root
+        mode: 420
+        path: /etc/chrony.conf
+  osImageURL: ""
+`
+
+func createChronyManifestContent(c *common.Cluster) (string, error) {
+	sources := make([]string, 0)
+
+	for _, host := range c.Hosts {
+		if swag.StringValue(host.Status) == models.HostStatusDisabled {
+			continue
+		}
+
+		var ntpSources []*models.NtpSource
+		if err := json.Unmarshal([]byte(host.NtpSources), &ntpSources); err != nil {
+			return "", err
+		}
+
+		for _, source := range ntpSources {
+			if source.SourceState == models.SourceStateSynced {
+				if !funk.Contains(sources, source.SourceName) {
+					sources = append(sources, source.SourceName)
+				}
+
+				break
+			}
+		}
+	}
+
+	content := defaultChronyConf[:]
+
+	for _, source := range sources {
+		content += "\n" + fmt.Sprintf("server %s iburst", source)
```

Comment:
since already using Sprintf, this should work:
```
content = fmt.Sprintf("%s\nserver %s iburst", content, source)
```

but if you find it less readable, maybe just move the newline:
```
content += fmt.Sprintf("\nserver %s iburst", source)
```